### PR TITLE
PlatformIO build instructions

### PR DESCRIPTION
**Pull request by probonopd**
Thanks @TheAssassin.

Reducing LOC is always a good thing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,42 +42,18 @@ Pre-built binaries are available for download on [GitHub Releases](https://githu
...
-HERE=$(readlink -f .)
-sed -i -e 's|#define SKETCH_VERSION ".*"|#define SKETCH_VERSION "'$VERSION'"|' $PWD/ESP8266WirelessPrintAsync/ESP8266WirelessPrintAsync.ino
-arduino --pref build.path=. --verify --verbose-build --board $BD ESP8266WirelessPrintAsync/ESP8266WirelessPrintAsync.ino
-BOARD=$(echo $BD | cut -d ":" -f 3)
-mv ./ESP8266WirelessPrintAsync.ino.bin "ESP8266WirelessPrintAsync_${BOARD}_${VERSION}.bin"
-
+# Get PlatformIO (the toolchain we use for compiling)
+git clone https://github.com/probonopd/WirelessPrinting
+cd WirelessPrinting
+wget -c https://downloads.egenix.com/python/install-pyrun
```
Comment (TheAssassin) [inline README.md]: There's no reason to use a (potentially) non-free Python distribution just to get PlatformIO running. Python 3 is usually available anyway. Even if `pip` was missing, you could just run `python3 -m ensurepip`, then run `pip` like `python3 -m pip install ...`.

Also, why even bother including this? A link to https://platformio.org/install/cli or https://docs.platformio.org/en/latest/installation.html is sufficient.

You're making this overly complicated.

File: README.md [hunk 2]
```diff
@@ -42,42 +42,18 @@ Pre-built binaries are available for download on [GitHub Releases](https://githu
...
+appdir/usr/bin/pip3 install -U platformio==4.2.1
+
+# Build the firmware (it downloads the needed libraries)
+appdir/usr/bin/platformio run
+ls -lh ./build_output/firmware/*.bin
 ```
+
 ### Flashing from Linux
 
 Can be flashed via USB or (subsequently) over the air. You can use the Arduino IDE if you compiled yourself, or one of the following commands if you just want to flash a precompiled firmware.
```
Comment (TheAssassin) [inline README.md]: Flashing is possible through `platformio run -t upload`, that's a lot easier than your commands there. I would at least mention this, even if you prefer people to use the Travis binaries.

File: README.md (hunk 1 above)
Comment (probonopd) [reply README.md]: I always want to boot any random (usually LTS) Linux distribution, and copy&paste something, and have it work. Without distribution-specific instructions that always seem to be for distributions I am not running at that moment. And without having to read up about how to install some auxiliary tools. From my experience, these instructions should be pretty much copy-and-paste-able on most systems. I love them.

File: README.md (hunk 2 above)
Comment (probonopd) [reply README.md]: Indeed. How do I specify which board is used, which port and so on? Last time I tried I found this less than intuitive in PIO (for newcomers)

File: README.md (hunk 1 above)
Comment (TheAssassin) [reply README.md]: It's highly confusing to random users, though. And bad style. Especially the use of a weird name like `appdir`. I think it's misleading. But it's your docs, I don't really care too much.

If you think PlatformIO's instructions are not sufficient, why not rather try to help there?

File: README.md (hunk 2 above)
Comment (TheAssassin) [reply README.md]: Oh, right, we have more than one environment, and I have not specified a default one in the config. That would be `platformio -e <env> -t upload`, where `env` could be either one in `platformio.ini` (e.g., `nodemcuv2`, `d1_mini` or `esp32dev` for most ESP32-based dev boards).

File: README.md (hunk 1 above)
Comment (probonopd) [reply README.md]: Agree that it's not really a full-fledged AppDir, hence will change the name...

File: README.md (hunk 2 above)
Comment (probonopd) [reply README.md]: We'd need this for e.g., `/dev/ttyUSBX` and for an IP address (OTA)...

Comment (TheAssassin) [reply README.md]: Only for your pre-built binaries. You can use PlatformIO to upload to either OTA and flash via any known flash method. See e.g., https://docs.platformio.org/en/latest/platforms/espressif8266.html#over-the-air-ota-update, https://docs.platformio.org/en/latest/platforms/espressif32.html#packages. It's all configurable.

Comment (probonopd) [reply README.md]: Actually, let's write that then.

**Pull request merged**